src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -4656,6 +4656,9 @@ void CodeGen::genCheckUseBlockInit()
             continue;
         }
 
+// TODO-Review: The code below is currently unreachable. We are guaranteed to execute one of the
+// 'continue' statements above.
+#if 0
```

Comment:
I don't want to hold up getting this merged - looks great BTW - but why didn't you just delete this?

Reply:
I want to go back and see if any of this needs to be resurrected. It looks like part of the unreachable code was supposed to be responsible for correctness. I'll clean this up in a subsequent PR.

Reply:
Perhaps open an issue for follow up on this, so we don't forget?

Reply:
I added a comment to the issue that tracks improving the heuristic: https://github.com/dotnet/runtime/issues/8890#issuecomment-617494895